dts/arm/yaml/st,stm32-usb.yaml

```diff
@@ -0,0 +1,47 @@
+---
+title: STM32 USB
+id: st,stm32-usb
+version: 0.1
+
+description: >
+    This binding gives a base representation of the STM32 USB controller
+
+inherits:
+  - !include usb.yaml
+
+properties:
+  - compatible:
+      constraint: "st,stm32-usb"
+
+  - reg:
+      type: array
+      description: mmio register space
+      generation: define
+      category: required
+
+  - interrupts:
+      type: array
+      category: required
+      description: required interrupts
+      generation: define
+
+  - interrupt-names:
+      type: stringlist
+      category: optional
+      description: readable string describing the interrupts
+      generation: define
+
+  - clocks:
+      type: array
+      category: required
+      description: Clock gate control information
+      generation: define
+
+  - ram-size:
```

Comment:
when you say the OTG datasheet, is that a STM32 datasheet or something USB spec?

Reply:
STM32 SOC's reference manual.